VisualRust/Downloader.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Runtime.Serialization;
+
+namespace VisualRust
+{
+    public class Downloader
+    {
+        private static String[] Sha256Fingerprints =
+        {
+            "0C:0B:AC:4C:96:D9:F2:2C:8D:7A:00:9F:2F:48:3D:7B:46:FE:2C:60:0B:52:19:5B:B4:80:47:36:7C:03:E9:41",
+            // all the known static.rust-lang.org cert fingerprints should be specified here. they expire every
+            // 2 years.
+        };
+
+        private static bool ValidateCertificate(Object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
+        {
+            if (sslPolicyErrors != SslPolicyErrors.None)
+            {
+                return false;
+            }
+
+            var cert2 = new X509Certificate2(certificate);
+            var time = System.DateTime.Now;
+
+            if (time > cert2.NotAfter || time < cert2.NotBefore)
```

Comment:
I can't find explicit documentation for it, but I see failures on common errors (self-signed: https://www.pcwebshop.co.uk/, wrong domain: https://tv.eurosport.com/, expired: https://testssl-expire.disig.sk/index.en.html). You will be getting `sslPolicyErrors != SslPolicyErrors.None` inside the callback in all those cases. This expiration check does nothing.